central/complianceoperator/v2/scanconfigurations/service/service_impl.go

```diff
@@ -131,3 +131,18 @@ func (s *serviceImpl) GetComplianceScanConfiguration(ctx context.Context, req *v
 
 	return convertStorageScanConfigToV2ScanStatus(ctx, scanConfig, s.complianceScanSettingsDS)
 }
+
+func (s *serviceImpl) GetComplianceScanConfigurationCount(ctx context.Context, request *v2.RawQuery) (*v2.ComplianceScanConfigurationCount, error) {
+	parsedQuery, err := search.ParseQuery(request.GetQuery(), search.MatchAllIfEmpty())
+	if err != nil {
+		return nil, errors.Wrap(errox.InvalidArgs, err.Error())
+	}
+	scanConfigs, err := s.complianceScanSettingsDS.GetScanConfigurations(ctx, parsedQuery)
```

Comment:
I think a better solution would be to:

- add the indexer to the datastore
- create a count in the datastore
- have the datastore count call the indexer
- have this service call the datastore count method.

The `store/index.go` already exists and has a `Count` that takes in a query.  We just need to wire up the datastore to use it.